src/Servers/Kestrel/Core/src/Internal/HttpConnection.cs

```diff
@@ -224,7 +207,7 @@ private HttpProtocols SelectProtocol()
                 error = CoreStrings.EndPointRequiresAtLeastOneProtocol;
             }
 
-            if (!http1Enabled && http2Enabled && hasTls && !Http2Id.Span.SequenceEqual(applicationProtocol.Span))
+            if (!http1Enabled && http2Enabled && hasTls && !Http2Id.SequenceEqual(applicationProtocol.Span))
```

Comment:
So this saves us the Http2Id allocation per connection if we don't use the Http2Id.

Reply:
It was always static this just moves it from code to data. The c# features stores it in the data section in the pe file and references it via the span